### ISSUE TITLE
train.py - Does it also train the backbone?

### ISSUE THREAD

**Issue by Leonhard1987** [fizyr/keras-retinanet#809]
First of all, thank you for the great work! 
I have the following problem:
I saw an argument called '--freeze-backbone', help='Freeze training of backbone layers.', action='store_true' within train.py.
This is clearly an indication that the backbone is also trained during a training by default.

However when reading the readme on github it says:
'The training procedure of keras-retinanet works with training models. These are stripped down versions compared to the inference model and only contains the layers necessary for training (regression and classification values).'

Could someone clarify what parts of the network are trained?

**Comment by hgaiser** [comment]
By default, all parts (except for batch normalization layers) are trained. The reference from the readme is talking about something different. For training purposes the output of the network is the raw output, but the "interpreted output" if you will, is not returned. When you change the network to an inference model, it takes the model with the raw output, "interprets" it, and wraps it in a new model.
